configurate-core/src/main/java/ninja/leaping/configurate/objectmapping/ObjectMapper.java

```diff
@@ -113,7 +127,12 @@ public void serializeTo(Object instance, ConfigurationNode node) throws ObjectMa
                 if (fieldVal == null) {
                     node.setValue(null);
                 } else {
-                    TypeSerializer serial = node.getOptions().getSerializers().get(this.fieldType);
+                    TypeSerializer serial = null;
+                    if (customSerializer != null) {
+                        serial = customSerializer;
+                    } else {
+                        serial = node.getOptions().getSerializers().get(this.fieldType);
+                    }
                     if (serial == null) {
                         throw new ObjectMappingException("No TypeSerializer found for field " + field.getName() + " of type " + this.fieldType);
                     }
```

Comment:
Very small nitpick, but this check can be moved inside the `else` block above, in the same way you did for the `deserializeFrom` method.